### ISSUE TITLE
Add weight decay optimizer

### ISSUE THREAD

**Issue by guillaumekln** [guillaumekln/OpenNMT-tf#16]
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/weight_decay_optimizers.py
